packages/react-refresh/src/ReactFreshBabelPlugin.js

```diff
@@ -689,16 +689,15 @@ export default function(babel, opts = {}) {
               return;
             }
             const handle = createRegistration(programPath, persistentID);
-            if (
-              (targetExpr.type === 'ArrowFunctionExpression' ||
-                targetExpr.type === 'FunctionExpression') &&
-              targetPath.parent.type === 'VariableDeclarator'
-            ) {
-              // Special case when a function would get an inferred name:
+            if (targetPath.parent.type === 'VariableDeclarator') {
```

Comment:
Seems like comment on line 708 is now outdated? What does the else code path match now?

Reply:
We add two references for a HOC, one inside and one outside. (findInnerComponents recurses with the same callback.) The else path still applies to the inside one. I'll update the comment.

Reply:
i.e. you end up with

```
let Foo = hoc(_c = () => {})
_c2 = Foo;
```

We could probably do more to preserve the names of inner components too but this should be 95% of the common code patterns.